sdk/security/keyvault/internal/challenge_policy.go

```diff
@@ -30,7 +31,12 @@ type KeyVaultChallengePolicyOptions struct {
 type keyVaultAuthorizer struct {
 	// tro is the policy's authentication parameters. These are discovered from an authentication challenge
 	// elicited ahead of the first client request.
-	tro                     policy.TokenRequestOptions
+	//
+	// Protected by troLock.
+	tro policy.TokenRequestOptions
+	// Lock protecting tro in case there are multiple concurrent initial requests.
+	troLock sync.Mutex
```

Comment:
```suggestion
	troLock sync.RWMutex
```
Would prefer to use the reader/writer mutex for better performance as the token request options are read more than they are written to

Reply:
Before we make any changes can we please confirm that this is still an issue with the latest version? From the bug report, the versions in use predate GA and I recall we made some changes in how authorization is called which might resolve this issue.

Reply:
Sure, I can make this change @gracewilcox.